onnxruntime/core/framework/execution_context.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+#include "core/common/logging/logging.h"
+#include "core/framework/ort_value.h"
+#include "core/framework/iexecutor.h"
+#include "core/framework/stream_handles.h"
+#include "core/graph/basic_types.h"
+#include "core/common/inlined_containers.h"
+#ifdef ENABLE_TRAINING
+#include "core/framework/partial_graph_execution_state.h"
+#endif
+
+namespace onnxruntime {
+class SessionState;
+class IExecutionFrame;
+class ExecutionFrame;
+
+using OrtValueIndex = int;
+
+struct ReleasePlan {
+  std::unique_ptr<std::atomic_int[]> value_ref_counts_;
+};
+
+class DeviceStreamCollectionImpl;
+class DeviceStreamCollection {
+ public:
+  DeviceStreamCollection(size_t num_streams, const SessionState& sess_state);
+  ~DeviceStreamCollection();
+  void SetDeviceStream(size_t, std::unique_ptr<Stream> stream);
+  void SetDeviceStream(size_t, Stream* stream);
+  const std::vector<Stream*>& GetStreams() const;
```

Comment:
can we return gsl::span<const Stream*> or does the caller need non-const access to the streams? if the latter this method probably shouldn't be const.